File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/injection/component/FragmentComponent.java
package com.national.security.community.injection.component;

import com.national.security.community.injection.module.FragmentModule;
import com.national.security.community.injection.scope.FragmentScope;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;

import dagger.Component;

@FragmentScope
@Component(dependencies = AppComponent.class, modules = FragmentModule.class)
public interface FragmentComponent {

    RxAppCompatActivity getActivity();

    // void inject(DailyFragment dailyFragment);

}

File: app/src/main/java/com/jmd_news_kotlin/di/component/FragmentComponent.kt
package com.jmd_news_kotlin.di.component

import android.app.Activity
import android.support.v4.app.Fragment
import com.jmd_news_kotlin.di.module.FragmentModule
import com.jmd_news_kotlin.di.scope.FragmentScope
import dagger.Component

@FragmentScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(FragmentModule::class))
interface FragmentComponent {
    fun getActivity(): Activity
    fun inject(fragment: Fragment)
}